doc/user/content/ops/deployment.md

```diff
@@ -12,61 +12,135 @@ issue](https://github.com/MaterializeInc/materialize/issues/new?labels=C-feature
 
 ## Memory
 
-Materialize stores the majority of its state in-memory, and works best when the streamed data
-can be reduced in some way. For example, if you know that only a subset of your rows and columns
-are relevant for your queries, it helps to avoid materializing sources or views until you've
-expressed this to the system (we can avoid stashing that data, which can in some cases dramatically
-reduce the memory footprint).
-
-To minimize the chances that Materialize runs out of memory in a production environment,
-we recommend you make additional memory available to Materialize via a SSD-backed
-swap file or swap partition.
-
-This is particularly important in Linux and in Docker, where swap may not be automatically
-setup for you:
+Materialize stores the majority of its state in memory, and works best when the
+streamed data can be reduced in some way. For example, if you know that only a
+subset of your rows and columns are relevant for your queries, it helps to avoid
+materializing sources or views until you've expressed this to the system.
+Materialize can then avoid stashing the full set of rows and columns, which can
+in some cases dramatically reduce Materialize's memory footprint.
+
+### Compaction
+
+To prevent memory from growing without bound, Materialize periodically
+"compacts" data in [arrangements](/overview/api-components#indexes). For
+example, if you have a source that tracks product inventory, you might receive
+periodic inventory updates throughout the day:
+
+```
+(T-shirts, 9:07am, +500)
+(T-shirts, 11:32am, -1)
+(T-shirts, 3:14pm, -2)
+```
+
+After logical compaction, Materialize will store only the current inventory
+state:
+
+```
+(T-shirts, 3:14pm, +497)
+```
+
+Materialize will only perform this compaction on data that falls outside the
+logical compaction window. The default compaction window is 60 seconds, but the
+window can be adjusted via the
+[`--logical-compaction-window`](/cli/#compaction-window) option.
+
+Adjusting the compaction window involves making a tradeoff between historical
+detail and memory usage. A larger compaction window retains more historical
+detail but uses more memory. A smaller compaction window uses less memory but
+also retains less historical detail.
+
+{{< warning >}}
+Setting the compaction window too small can prevent Materialize from finding a
+suitable timestamp at which to execute a query, especially if that query joins
+multiple sources together. For best results, use a compaction window of at least
+five seconds.
+{{< /warning >}}
+
+Note that compaction only occurs in response to updates arriving. As a result,
```

Comment:
I don't want to complain about the text, which I think is fine at this level of detail, but did want to flag that this isn't exactly accurate. The `DIFFERENTIAL_EAGER_MERGE` parameter allows physical compaction to occur without updates arriving (and physical compaction is when logical compaction occurs). The problem is that thes does not apply once the underlying data have been fully physically merged. If there is only one physical batch, the parameter does not introduce speculative merging that would be necessary to pick up logical compaction that becomes possible after physical compaction.

Reply:
Ack, rephrased slightly to be less wrong without going into more detail.